In [1]:
import sys
import os
import logging

# To‘g‘ri yo‘lni sys.path ga qo‘shish
sys.path.append(r"C:\Users\Rasulbek907\Desktop\Project_8\Source")

# Fayl nomi: Auto_Pipeline.py, shuning uchun import ham shunday bo'lishi kerak
from pipeline import DataLoader

# Log fayl manzili
log_path = r"C:\Users\Rasulbek907\Desktop\Project_8\Log\auto_pipeline.log"

# Log sozlamalari
logging.basicConfig(
    filename=log_path,
    filemode='a',
    format='%(asctime)s - %(levelname)s - %(message)s',
    level=logging.INFO
)

# Ma'lumot faylining manzili
data_path = r"C:\Users\Rasulbek907\Desktop\Project_8\Data\Feature_Selection\Filtered_Features.csv"

try:
    logging.info("Ma'lumot fayli yuklanmoqda...")
    df = DataLoader.load_csv(data_path)
    logging.info(f"Fayl muvaffaqiyatli yuklandi. Satrlar soni: {len(df)}, ustunlar soni: {len(df.columns)}")
except Exception as e:
    logging.error(f"Ma'lumot faylini yuklashda xatolik yuz berdi: {e}")

In [2]:
import sys
import os
import importlib

# 1. path qo‘shish
source_path = r'C:\Users\Rasulbek907\Desktop\amaliyot\Source'
if source_path not in sys.path:
    sys.path.append(source_path)

# 2. Import va reload
import pipeline
importlib.reload(pipeline)  # 🔄 MUHIM: o‘zgargan faylni yangilaydi

# 3. DataLoader ni olish
from pipeline import DataLoader

# 4. Foydalanish
data_path = r'C:\Users\Rasulbek907\Desktop\Project_8\Data\Feature_Selection\Filtered_Features.csv'
loader = DataLoader(data_path)

# 5. Endi ishlaydi!
df = loader.load_dataset()
print(df.head()) 

   Unnamed: 0     index  dateCrawled      name  vehicleType  \
0           0  0.000000     0.584454  0.015422     0.428571   
1           1  0.000003     0.273117  0.391083     1.000000   
2           2  0.000005     0.378148  0.332203     0.571429   
3           3  0.000008     0.823166  0.733871     0.571429   
4           4  0.000011     0.965794  0.119477     0.857143   

   yearOfRegistration     model  monthOfRegistration  fuelType     brand  \
0            0.724138  0.475806             0.416667  0.500000  0.025641   
1            0.482759  0.479839             0.666667  0.500000  0.358974   
2            0.379310  0.475806             0.500000  0.166667  0.974359   
3            0.620690  0.415323             0.583333  0.500000  0.794872   
4            0.172414  0.044355             0.833333  0.166667  0.051282   

   postalCode  lastSeen  abtest_control  gearbox_automatik  \
0    0.665989  0.972885             0.0                0.0   
1    0.903792  0.879404             0.0 

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 322278 entries, 0 to 322277
Data columns (total 16 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   Unnamed: 0            322278 non-null  int64  
 1   index                 322278 non-null  float64
 2   dateCrawled           322278 non-null  float64
 3   name                  322278 non-null  float64
 4   vehicleType           322278 non-null  float64
 5   yearOfRegistration    322278 non-null  float64
 6   model                 322278 non-null  float64
 7   monthOfRegistration   322278 non-null  float64
 8   fuelType              322278 non-null  float64
 9   brand                 322278 non-null  float64
 10  postalCode            322278 non-null  float64
 11  lastSeen              322278 non-null  float64
 12  abtest_control        322278 non-null  float64
 13  gearbox_automatik     322278 non-null  float64
 14  notRepairedDamage_ja  322278 non-null  float64
 15  

In [4]:
import sys
import os
import importlib

source_path = r'C:\Users\Rasulbek907\Desktop\amaliyot\Source'
if source_path not in sys.path:
    sys.path.append(source_path)

import pipeline
importlib.reload(pipeline)  

from auto_pipeline import Auto_Pipeline

# Avto Pipeline RandomForestRegressor

In [5]:
from sklearn.ensemble import RandomForestRegressor

pipeline_rf = Auto_Pipeline(df=df, target="kilometer_cont", model=RandomForestRegressor(n_estimators=100))
pipeline_rf.fit()
y_pred = pipeline_rf.predict()

# R2 ballini olish
r2_score = pipeline_rf.score()
print("R2 score:", r2_score)

R2 score: 0.9307196396683416


# Avto Pipeline DecisionTreeRegressor

In [6]:
from sklearn.tree import DecisionTreeRegressor

pipeline_dt = Auto_Pipeline(df=df, target="kilometer_cont", model=DecisionTreeRegressor())
pipeline_dt.fit()
y_pred = pipeline_dt.predict()

r2_score = pipeline_dt.score()
print("Decision Tree R2 score:", r2_score)

Decision Tree R2 score: 1.0


# Avto Pipeline LinearRegression

In [8]:
from sklearn.linear_model import LinearRegression

pipeline_lr = Auto_Pipeline(df=df, target="kilometer_cont", model=LinearRegression())
pipeline_lr.fit()
y_pred = pipeline_lr.predict()

r2_score = pipeline_lr.score()
print("Linear Regression  R2 score:", r2_score)

Linear Regression  R2 score: 0.3672211260123365


In [17]:
import joblib
import os

model_path = r"C:\Users\Rasulbek907\Desktop\Project_8\Models\Pipeline_Models\LinearRegression.joblib"
os.makedirs(os.path.dirname(model_path), exist_ok=True)

joblib.dump(pipeline_lr, model_path)
print(f"✅ Model muvaffaqiyatli saqlandi: {model_path}")

✅ Model muvaffaqiyatli saqlandi: C:\Users\Rasulbek907\Desktop\Project_8\Models\Pipeline_Models\LinearRegression.joblib


# Natijalarni Taqqoslash 

# Auto Pipeline Regression Results

| Model                     | R2 Score |
|---------------------------|----------|
| **Decision Tree**         | <span style="color:green">1.0</span> |
| Random Forest             | <span style="color:orange">0.93</span> |
| Linear Regression         | <span style="color:red">0.36</span> |
